rancher2/resource_rancher2_cluster_v2.go

```diff
@@ -165,9 +165,13 @@ func clusterV2StateRefreshFunc(meta interface{}, objID string) resource.StateRef
 	return func() (interface{}, string, error) {
 		obj, err := getClusterV2ByID(meta.(*Config), objID)
 		if err != nil {
-			if IsNotFound(err) || IsForbidden(err) || IsNotLookForByID(err) {
+			if IsNotFound(err) || IsForbidden(err) {
 				return obj, "removed", nil
 			}
+			// This is required to allow standard user to use this resource
+			if IsNotAccessibleByID(err) {
```

Comment:
If a user cannot access something by ID, then they should not be given access to that item. This seems like privilege escalation.

I think it should be determined why the user doesn't have access to it and fix the permissions.

Reply:
Ah I see - I thought Terraform should provision anyway despite a standard user not being able to access the machine config. I'll look into this a little more

Reply:
Here's the new PR https://github.com/rancher/rancher/pull/37230